nni/algorithms/compression/v2/pytorch/pruning/basic_pruner.py

```diff
@@ -440,12 +441,12 @@ def reset_tools(self):
         else:
             self.data_collector.reset()
         if self.metrics_calculator is None:
-            self.metrics_calculator = DistMetricsCalculator(p=2, dim=0)
+            self.metrics_calculator = DistMetricsCalculator(p=2, scalors=Scaling(kernel_size=[1], kernel_padding_mode='back'))
         if self.sparsity_allocator is None:
             if self.mode == 'normal':
-                self.sparsity_allocator = NormalSparsityAllocator(self, dim=0)
+                self.sparsity_allocator = NormalSparsityAllocator(self, Scaling(kernel_size=[1], kernel_padding_mode='back'))
             elif self.mode == 'dependency_aware':
-                self.sparsity_allocator = Conv2dDependencyAwareAllocator(self, 0, self.dummy_input)
+                self.sparsity_allocator = DependencyAwareAllocator(self, self.dummy_input)
```

Comment:
`DependencyAwareAllocator` does not need `scalors`?

Reply:
only channel pruning is meaningful in `DependencyAwareAllocator`, but you are right, the `scalors` should be exposed in the interface.